core/src/main/java/com/graphhopper/routing/lm/LandmarkStorage.java

```diff
@@ -0,0 +1,745 @@
+        return new BBox(-20, 179, -50, 85);
+    }
+
+    /**
+     * This method makes edges crossing the specified border inaccessible to split a bigger area into smaller subnetworks.
+     * Important for the world wide use case to limit the maximum distance but also to faster detect unreasonable routes.
+     */
+    protected int splitCountries(Map<String, Polygon> borderMap, SpatialRuleLookupArray ruleLookup) {
+        if (borderMap.isEmpty())
+            return 0;
+
+        for (final Map.Entry<String, Polygon> polgonEntry : borderMap.entrySet())
+            ruleLookup.addRule(new DefaultSpatialRule() {
+                @Override
+                public String getCountryIsoA3Name() {
+                    return polgonEntry.getKey();
+                }
+            }.addBorder(polgonEntry.getValue()));
+
+        AllEdgesIterator allEdgesIterator = graph.getAllEdges();
+        NodeAccess nodeAccess = graph.getNodeAccess();
+        int inaccessible = 0;
+        while (allEdgesIterator.next()) {
+            int adjNode = allEdgesIterator.getAdjNode();
+            SpatialRule ruleAdj = ruleLookup.lookupRule(nodeAccess.getLatitude(adjNode), nodeAccess.getLongitude(adjNode));
+
+            int baseNode = allEdgesIterator.getBaseNode();
+            SpatialRule ruleBase = ruleLookup.lookupRule(nodeAccess.getLatitude(baseNode), nodeAccess.getLongitude(baseNode));
+            // TODO cache rule for all nodes to avoid calling "2*nodes" of lookup
+            if (ruleAdj != ruleBase) {
+                inaccessible++;
+                allEdgesIterator.setFlags(encoder.setAccess(allEdgesIterator.getFlags(), false, false));
+            }
+        }
+        return inaccessible;
+    }
+
+    private String myprint(int... ints) {
+        EdgeFilter outEdgeFilter = new DefaultEdgeFilter(encoder, false, true);
+        EdgeFilter inEdgeFilter = new DefaultEdgeFilter(encoder, true, false);
+        String str = "";
+
+        for (int node : ints) {
+            if (node < graph.getNodes()) {
+                str += "out " + GHUtility.getNodeInfo(graph, node, outEdgeFilter) + "\n\n";
+                str += "in  " + GHUtility.getNodeInfo(graph, node, inEdgeFilter) + "\n\n";
+            }
+        }
+        System.out.println(str);
+        return str;
+    }
+
+    /**
+     * This method creates landmarks for the specified subnetwork (integer list)
+     *
+     * @return landmark mapping
+     */
+    private boolean createLandmarks(final int startNode, final byte[] subnetworks, IntArrayList subnetworkIds) {
+        final int subnetworkId = landmarkIDs.size();
+
+        // 1a) pick landmarks via shortest weighting for a better geographical spreading
+        // 'fastest' has big problems with ferries (slow&very long) and allowing arbitrary weighting is too dangerous
+        Weighting initWeighting = lmSelectionWeighting;
+        Explorer explorer = new Explorer(graph, this, initWeighting, traversalMode);
```

Comment:
Thought at first this would be the EdgeExplorer. Maybe explicitly call this LandmarkExplorer?